litert/vendors/intel_openvino/dispatch/device_context.cc

```diff
@@ -0,0 +1,200 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "litert/vendors/intel_openvino/dispatch/device_context.h"
+#if __ANDROID__
+#include <android/hardware_buffer.h>
+#endif  // __ANDROID__
+
+#include <string.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include <openvino/runtime/intel_npu/level_zero/level_zero.hpp>
+#include <openvino/runtime/remote_context.hpp>
+
+#include "litert/c/litert_common.h"
+#include "litert/c/litert_model.h"
+
+litert::Expected<LiteRtDispatchDeviceContextT::Ptr>
+LiteRtDispatchDeviceContextT::Create(ov::Core core) {
+  return Ptr(new LiteRtDispatchDeviceContextT(core));
+}
+
+#if LITERT_HAS_AHWB_SUPPORT
+litert::Expected<int> GetFdFromUnixHandle(AHardwareBuffer *ahwb) {
+  int socks[2];
+  if (socketpair(AF_UNIX, SOCK_SEQPACKET, 0, socks) != 0) {
+    return litert::Unexpected(kLiteRtStatusErrorRuntimeFailure,
+                              "Failed to create socket pair");
+  }
+  if (AHardwareBuffer_sendHandleToUnixSocket(ahwb, socks[0]) != 0) {
```

Comment:
nit: You may want to define absl::Cleanup to close the created sockets.
https://github.com/abseil/abseil-cpp/tree/master/absl/cleanup